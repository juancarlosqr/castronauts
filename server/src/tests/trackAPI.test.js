const createServer = require('./utils/createServer');
const {
  GET_LAUNCHES,
  GET_LAUNCHES_AUTHOR,
  GET_LAUNCHES_MODULES,
  GET_TRACK,
} = require('./utils/queries');

describe('trackAPI DataSource', () => {
  let server;

  beforeEach(async () => {
    server = createServer();
  });

  it('fetches tracks', async () => {
    // run operation and snapshot the output
    const res = await server.executeOperation({ query: GET_LAUNCHES });
    expect(res).toMatchSnapshot();
  });

  it('fetches tracks with author', async () => {
    // run operation and snapshot the output
    const res = await server.executeOperation({ query: GET_LAUNCHES_AUTHOR });
    expect(res).toMatchSnapshot();
  });

  it('fetches tracks with modules', async () => {
    // run operation and snapshot the output
    const res = await server.executeOperation({ query: GET_LAUNCHES_MODULES });
    expect(res).toMatchSnapshot();
  });

  it('fetches single track', async () => {
    // run operation and snapshot the output
    const res = await server.executeOperation({
      query: GET_TRACK,
      variables: { trackId: 'any' }, // not sure how to pass right trackId
    });
    expect(res).toMatchSnapshot();
  });
});
