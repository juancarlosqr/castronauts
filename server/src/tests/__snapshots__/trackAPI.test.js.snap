// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trackAPI DataSource fetches tracks 1`] = `
Object {
  "data": Object {
    "tracksForHome": Array [
      Object {
        "description": "Est quod et suscipit distinctio tempore rerum sunt atque.",
        "id": "4a1a1ce9-0544-4a7e-80d3-b509dc39e69e",
        "length": 7,
        "modulesCount": 2,
        "numberOfViews": 51,
        "thumbnail": "http://www.Sadye.ca/",
        "title": "Hic aut ad maxime.",
      },
      Object {
        "description": "Qui sed repellendus minus.",
        "id": "c40ddac6-f18c-4f3e-9f99-50e24122d8c8",
        "length": 3,
        "modulesCount": 4,
        "numberOfViews": 2,
        "thumbnail": "http://www.Little.tv/",
        "title": "Ut aut sed vero sint velit et et.",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`trackAPI DataSource fetches tracks with author 1`] = `
Object {
  "data": Object {
    "tracksForHome": Array [
      Object {
        "author": Object {
          "id": "4cf3e745-d2ad-4885-ab28-f90277a2703f",
          "name": "Dr. Jacklyn Skiles",
          "photo": "http://Farrell.info/",
        },
        "id": "4a1a1ce9-0544-4a7e-80d3-b509dc39e69e",
        "length": 7,
        "modulesCount": 2,
        "thumbnail": "http://www.Sadye.ca/",
        "title": "Hic aut ad maxime.",
      },
      Object {
        "author": Object {
          "id": "4cf3e745-d2ad-4885-ab28-f90277a2703f",
          "name": "Dr. Jacklyn Skiles",
          "photo": "http://Farrell.info/",
        },
        "id": "c40ddac6-f18c-4f3e-9f99-50e24122d8c8",
        "length": 3,
        "modulesCount": 4,
        "thumbnail": "http://www.Little.tv/",
        "title": "Ut aut sed vero sint velit et et.",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
